src/math/Bspline.cpp

```diff
@@ -28,23 +29,29 @@ Bspline::Bspline(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int splineDegree
   //The code for computing the knots and the control points is copied from Eigens bspline interpolation with minor modifications, https://gitlab.com/libeigen/eigen/-/blob/master/unsupported/Eigen/src/Splines/SplineFitting.h
 
   Eigen::KnotAveraging(ts, splineDegree, _knots);
-  Eigen::DenseIndex n = xs.cols();
-  Eigen::MatrixXd   A = Eigen::MatrixXd::Zero(n, n);
+  Eigen::DenseIndex           n = xs.cols();
+  Eigen::SparseMatrix<double> A(n, n);
 
+  std::vector<Eigen::Triplet<double>> matrixEntries;
```

Comment:
Given that this is a big vector, and we know the amount of elements that will be added, I am positive that reserving space will make a significant performance difference as it prevent reallocations.

```cpp
matrixEntries.reserve( /* num elements from loop n*splineDegree? */);
```